### PR TITLE
Fix rank 2 batch norm

### PR DESCRIPTION
**Pull request by CharlieL7**
* Allows for rank 2 tensors into batchnorm
  * specifically when spatial dimensions are all 1 and removed

### CODE REVIEW

**Review by umangyadav — APPROVED** [review]

File: src/onnx/parse_batchnorm.cpp [hunk 1]
```diff
@@ -54,15 +54,16 @@ struct parse_batchnorm : op_parser<parse_batchnorm>
             MIGRAPHX_THROW("PARSE_BATCHNORM: argument scale, bias, mean, or var rank != 1");
         }
 
-        if(x_lens.size() == 1)
+        if(x_lens.size() == 1 or x_lens.size() == 2)
```
Comment (kahmed10) [inline src/onnx/parse_batchnorm.cpp:57]:
```suggestion
        auto x_rank = x_lens.size();
        // can get a rank 2 tensor when trailing 1 dims are removed, ex: 1x1024x1x1 - > 1x1024
        if(x_rank == 1 or x_rank == 2)
```

File: src/onnx/parse_batchnorm.cpp [hunk 2]
```diff
@@ -54,15 +54,16 @@ struct parse_batchnorm : op_parser<parse_batchnorm>
...
         {
-            auto rt   = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.5}});
-            auto eps  = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {epsilon}});
-            auto n0   = info.add_broadcastable_binary_op("sub", args[0], args[3]);
-            auto d0   = info.add_broadcastable_binary_op("add", args[4], eps);
-            auto d1   = info.add_broadcastable_binary_op("pow", d0, rt);
-            auto div0 = info.add_broadcastable_binary_op("div", n0, d1);
-            auto r0   = info.add_broadcastable_binary_op("mul", div0, args[1]);
+            // can get a rank 2 tensor when trailing 1 dims are removed, ex: 1x1024x1x1 - > 1x1024
```
Comment (kahmed10) [inline src/onnx/parse_batchnorm.cpp:59]:
```suggestion
```

**Review by kahmed10 — APPROVED** [review]
